### PR TITLE
Switch update logic to use Download Center based metadata links

### PR DESCRIPTION
**Pull request by caohai**
This PR updates the ADS update logic to directly fetch build metadata files and parse new build info without querying the dedicated update server.

- The metadata files contains build info (version, build download url in Download Center, etc) which we previously used cosmos db to store.
- The metadata files are statically hosted in the Download Center and we use a static FWLink to point to the latest version.
- `macOS` version update requires separate metadata files because the update logic uses the built-in `autoUpdater` from electron, which has specific requirements on the response from the feed url. These macOS metadata files contains the same content as the response from the current update server.

Verification:
- We can verify this update path by creating a build from the PR branch with a lower version 1.47.0 and test the update to the latest 1.48.0.
- I've queued a build at https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=226610&view=results and will add manual verification results to the PR later.

Packet capture for the update process: the 1st call is for checking if the current version is the latest, this happens during the initial bootup, the 2nd call is metadata retrieving and the 3rd call is for downloading the actual build,
image

### CODE REVIEW

File: src/vs/platform/update/electron-main/updateMetadataProvider.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+export interface Build {
+	id: string;
+	version: string;
+	isFrozen?: boolean;
+	assets: Asset[];
+	updates: { [platform: string]: string; };
+}
+
+export function getUpdateFromBuild(build: Build | null, productService: IProductService, platform: string): IUpdate | undefined {
```
Comment (caohai) [inline src/vs/platform/update/electron-main/updateMetadataProvider.ts:36]: This logic is copied from the update server. We are basically moving metadata parsing logic to the client side here.

File: src/vs/platform/update/electron-main/updateMetadataProvider.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// {{SQL CARBON EDIT}} - This file contains interfaces, util functions and url definitions
+// to support Download Center based update.
+import { IProductService } from 'vs/platform/product/common/productService';
+import { IUpdate } from 'vs/platform/update/common/update';
+
+export const updateMetadataUrl = 'https://go.microsoft.com/fwlink/?linkid=2274438';
```
Comment (Benjin) [inline src/vs/platform/update/electron-main/updateMetadataProvider.ts:11]: Can you add a comment explaining why there are different URLs for each Mac target, but Windows and Linux are combined?

**Review by Benjin — APPROVED** [review]

File: src/vs/platform/update/electron-main/updateMetadataProvider.ts (hunk 2 above)
Comment (Charles-Gagnon) [inline src/vs/platform/update/electron-main/updateMetadataProvider.ts:11]: suggestion: Consider having these values be defined in product.json instead so they can be updated without requiring a full rebuild

Comment (caohai) [reply src/vs/platform/update/electron-main/updateMetadataProvider.ts]: moved them to product.json

Comment (caohai) [reply src/vs/platform/update/electron-main/updateMetadataProvider.ts]: added comment in the field definition in `product.ts`

**Review by Charles-Gagnon — APPROVED** [review]

File: product.json [hunk 3]
```diff
@@ -47,6 +47,10 @@
 	"gettingStartedUrl": "https://go.microsoft.com/fwlink/?linkid=862039",
 	"releaseNotesUrl": "https://go.microsoft.com/fwlink/?linkid=875578",
 	"documentationUrl": "https://go.microsoft.com/fwlink/?linkid=862277",
+	"updateMetadataUrl": "https://go.microsoft.com/fwlink/?linkid=2274438",
+	"updateMetadataMacUrl": "https://go.microsoft.com/fwlink/?linkid=22745",
```
Comment (Charles-Gagnon) [inline product.json:51]: The mac links all seem to be not valid - is that expected?

Comment (caohai) [reply product.json]: Thanks for catching this! I accidentally deleted two characters when doing the bulk edit. Updated to the correct links and verified the update url via builds here:
https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=226674&view=results

**Pull request merged**